classes/template/PKPTemplateManager.inc.php

```diff
@@ -249,49 +327,297 @@ function setCacheability($cacheability = CACHEABILITY_PUBLIC) {
+			$this->_styleSheets[$context][$args['priority']][$name] = array(
+				'style' => $style,
+				'inline' => $args['inline'],
+			);
+		}
+	}
+
+	/**
+	 * Register a script with the script handler
+	 *
+	 * @param $name string Unique name for the script
+	 * @param $script string The script to be included. Should be a URL or, if
+	 *   the `inline` argument is included, script data to be output.
+	 * @param $args array Key/value array defining display details
+	 *   `priority` int The order in which to print this script.
+	 *      Default: STYLE_SEQUENCE_NORMAL
+	 *   `contexts` string|array Where the script should be loaded.
+	 *      Default: array('frontend')
+	 *   `inline` bool Whether the $script value should be output directly as
+	 *      script data. Used to pass backend data to the scripts.
+	 */
+	function addJavaScript($name, $script, $args = array()) {
+
+		$args = array_merge(
+			array(
+				'priority' => STYLE_SEQUENCE_NORMAL,
+				'contexts' => array('frontend'),
+				'inline'   => false,
+			),
+			$args
+		);
+
+		$args['contexts'] = (array) $args['contexts'];
+		foreach($args['contexts'] as $context) {
+			$this->_javaScripts[$context][$args['priority']][$name] = array(
+				'script' => $script,
+				'inline' => $args['inline'],
+			);
+		}
+	}
+
+	/**
+	 * Add a page-specific item to the <head>.
+	 *
+	 * @param $name string Unique name for the header
+	 * @param $header string The header to be included.
+	 * @param $args array Key/value array defining display details
+	 *   `priority` int The order in which to print this header.
+	 *      Default: STYLE_SEQUENCE_NORMAL
+	 *   `contexts` string|array Where the header should be loaded.
+	 *      Default: array('frontend')
+	 */
+	function addHeader($name, $header, $args = array()) {
+
+		$args = array_merge(
+			array(
+				'priority' => STYLE_SEQUENCE_NORMAL,
+				'contexts' => array('frontend'),
+			),
+			$args
+		);
+
+		$args['contexts'] = (array) $args['contexts'];
+		foreach($args['contexts'] as $context) {
+			$this->_htmlHeaders[$context][$args['priority']][$name] = array(
+				'header' => $header,
+			);
+		}
+	}
+
+	/**
+	 * Register all files required by the core JavaScript library
+	 */
+	function registerJSLibrary() {
+
+		$basePath = $this->_request->getBasePath();
+		$baseUrl = $this->_request->getBaseUrl();
+
+		// Load minified file if it exists
+		if (Config::getVar('general', 'enable_minified')) {
+			$path = $basePath . '/js/pkp.min.js';
+			if (file_exists($path)) {
+				$this->addJavaScript(
+					'pkpLib',
+					$path,
+					array(
+						'priority' => STYLE_SEQUENCE_CORE,
+						'contexts' => array('backend', 'frontend')
+					)
+				);
+				return;
 			}
+		}
+
+		// Otherwise retrieve and register all script files
+		$minifiedScriptsTemplate = $this->fetch('common/minifiedScripts.tpl');
+		preg_match_all('/<script src=\"' . str_replace('/', '\/', $baseUrl ) . '(.*)\"><\/script>/', $minifiedScriptsTemplate, $scripts);
```

Comment:
Use `preg_quote` rather than manually escaping the slashes. Also, I'd recommend using `[^"]* rather than `.*` to decrease the chances of an overzealous match.